tests/TestCase.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace React\Tests\Dns;
+
+use React\Promise\Deferred;
+
+abstract class TestCase extends \PHPUnit_Framework_TestCase
+{
+    protected function expectCallableOnce()
+    {
+        $mock = $this->createCallableMock();
+        $mock
+            ->expects($this->once())
+            ->method('__invoke');
+
+        return $mock;
+    }
+
+    protected function expectCallableOnceWith($value)
+    {
+        $mock = $this->createCallableMock();
+        $mock
+            ->expects($this->once())
+            ->method('__invoke')
+            ->with($this->equalTo($value));
+
+        return $mock;
+    }
+
+    protected function expectCallableNever()
+    {
+        $mock = $this->createCallableMock();
+        $mock
+            ->expects($this->never())
+            ->method('__invoke');
+
+        return $mock;
+    }
+
+    protected function createCallableMock()
+    {
+        return $this->getMock('React\Tests\Dns\CallableStub');
+    }
+
+    protected function createPromiseResolved($value = null)
+    {
+        $deferred = new Deferred();
+        $deferred->resolve($value);
+        return $deferred->promise();
+    }
+
+    protected function createPromiseRejected($value = null)
+    {
+        $deferred = new Deferred();
+        $deferred->reject($value);
+        return $deferred->promise();
+    }
+}
```

Comment:
These 2 methods can be shortened to:

``` php
    protected function createPromiseResolved($value = null)
    {
        return Promise\resolve($value);
    }

    protected function createPromiseRejected($value = null)
    {
        return Promise\reject($value);
    }
```

which makes them kind of obsolete ;)